subprojects/ide-native/src/main/java/org/gradle/ide/xcode/tasks/GenerateSchemeFileTask.java

```diff
@@ -54,28 +57,47 @@ protected void configure(XcodeSchemeFile schemeFile) {
     }
 
     private void configureBuildAction(XcodeSchemeFile.BuildAction action) {
-        action.entry(new Action<XcodeSchemeFile.BuildActionEntry>() {
-            @Override
-            public void execute(XcodeSchemeFile.BuildActionEntry buildActionEntry) {
-                buildActionEntry.setBuildForAnalysing(true);
-                buildActionEntry.setBuildForArchiving(true);
-                buildActionEntry.setBuildForProfiling(true);
-                buildActionEntry.setBuildForRunning(true);
-                buildActionEntry.setBuildForTesting(true);
-                buildActionEntry.setBuildableReference(toBuildableReference(xcodeProject.getTarget()));
-            }
-        });
+        for (final XcodeTarget xcodeTarget : xcodeProject.getTargets()) {
+            action.entry(new Action<XcodeSchemeFile.BuildActionEntry>() {
+                @Override
+                public void execute(XcodeSchemeFile.BuildActionEntry buildActionEntry) {
+                    buildActionEntry.setBuildForAnalysing(!PBXTarget.ProductType.UNIT_TEST.equals(xcodeTarget.getProductType()));
+                    buildActionEntry.setBuildForArchiving(!PBXTarget.ProductType.UNIT_TEST.equals(xcodeTarget.getProductType()));
+                    buildActionEntry.setBuildForProfiling(!PBXTarget.ProductType.UNIT_TEST.equals(xcodeTarget.getProductType()));
+                    buildActionEntry.setBuildForRunning(!PBXTarget.ProductType.UNIT_TEST.equals(xcodeTarget.getProductType()));
+                    buildActionEntry.setBuildForTesting(PBXTarget.ProductType.UNIT_TEST.equals(xcodeTarget.getProductType()));
+                    buildActionEntry.setBuildableReference(toBuildableReference(xcodeTarget));
+                }
+            });
+        }
     }
 
     private void configureTestAction(XcodeSchemeFile.TestAction action) {
         action.setBuildConfiguration("Debug");
+
+        for (final XcodeTarget xcodeTarget : xcodeProject.getTargets()) {
+            if (PBXTarget.ProductType.UNIT_TEST.equals(xcodeTarget.getProductType())) {
+                action.setBuildConfiguration("__GradleTestRunner_Debug");
```

Comment:
Does this name have some kind of special meaning? IOW, why wouldn't it just be "Debug" or "Test Debug" or something like that?